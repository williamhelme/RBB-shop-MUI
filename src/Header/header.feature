Feature: Header

Scenario: Loading the App header
  Given the header has loaded
  Then there should be a header tag

Scenario: Should skew the background
  Given a skew property is provided
  Then the background should skew

Scenario: it renders correctly
  Given it has rendered
  Then it should match the snapshot