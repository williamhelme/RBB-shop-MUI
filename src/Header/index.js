import React, { Component } from "react";

class Header extends Component {
  render() {
    const { skew = 20 } = this.props;
    return (
      <header>
        <object
          type="image/svg+xml"
          data="./brand_logo.svg"
          style={{
            position: "absolute",
            height: "10em",
            left: "2%",
            top: "2%",
            zIndex: 1
          }}
        >
          <img src="./cake.jpg" alt="No SVG support" />
        </object>
        <section
          style={{
            height: "40vh",
            width: "100vw",
            position: "relative"
          }}
        >
          <article
            style={{
              height: "inherit",
              background: "url(./cake.jpg) no-repeat",
              backgroundPosition: "center",
              backgroundSize: "cover",
              filter: "blur(1px)"
            }}
          />
          <svg
            style={{
              position: "absolute",
              bottom: -3,
              left: -100,
              width: "calc(100vw + 101px)", // extra width to stop blur on sides
              height: `${skew}%`,
              color: "var(--main-bg-color)",
              zIndex: 1
            }}
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 100 100"
            preserveAspectRatio="none"
          >
            <polygon className="svg-header" points="0,100 100,0 100,100" />
          </svg>
        </section>
      </header>
    );
  }
}

export default Header;
