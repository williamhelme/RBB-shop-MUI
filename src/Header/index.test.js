import React from 'react'
import { loadFeature, defineFeature } from 'jest-cucumber';
import { shallow } from 'enzyme'
import Header from './'

const feature = loadFeature('./src/Header/header.feature');

defineFeature(feature, test => {
  test('Loading the App header', ({ given, when, then, pending }) => {
    let Elem;
    
    given('the header has loaded', () => {
      Elem = shallow(<Header />);
    })
    
    then('there should be a header tag', () => {
      expect(Elem.type()).toEqual('header');
    })
  })

  test('Should skew the background', ({ given, when, then, pending }) => {
    let Elem;
    given('a skew property is provided', () => {
      Elem = shallow(<Header skew={6} />);
    });

    then('the background should skew', () => {
      expect(Elem.find('section').length).toEqual(1);
      
      expect(Elem.find('article').length).toEqual(1);
    });
  });

})