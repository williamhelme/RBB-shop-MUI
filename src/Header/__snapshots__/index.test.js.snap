// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header it renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <object
          data="./brand_logo.svg"
          style={
            Object {
              "height": "10em",
              "left": "2%",
              "position": "absolute",
              "top": "2%",
              "zIndex": 1,
            }
          }
          type="image/svg+xml"
        >
          <img
            alt="No SVG support"
            src="./cake.jpg"
          />
        </object>,
        <section
          style={
            Object {
              "height": "40vh",
              "position": "relative",
              "width": "100vw",
            }
          }
        >
          <article
            style={
              Object {
                "background": "url(./cake.jpg) no-repeat",
                "backgroundPosition": "center",
                "backgroundSize": "cover",
                "filter": "blur(1px)",
                "height": "inherit",
              }
            }
          />
          <svg
            preserveAspectRatio="none"
            style={
              Object {
                "bottom": -3,
                "color": "var(--main-bg-color)",
                "height": "20%",
                "left": -100,
                "position": "absolute",
                "width": "calc(100vw + 101px)",
                "zIndex": 1,
              }
            }
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              className="svg-header"
              points="0,100 100,0 100,100"
            />
          </svg>
        </section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="No SVG support"
            src="./cake.jpg"
          />,
          "data": "./brand_logo.svg",
          "style": Object {
            "height": "10em",
            "left": "2%",
            "position": "absolute",
            "top": "2%",
            "zIndex": 1,
          },
          "type": "image/svg+xml",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "No SVG support",
            "src": "./cake.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "object",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <article
              style={
                Object {
                  "background": "url(./cake.jpg) no-repeat",
                  "backgroundPosition": "center",
                  "backgroundSize": "cover",
                  "filter": "blur(1px)",
                  "height": "inherit",
                }
              }
            />,
            <svg
              preserveAspectRatio="none"
              style={
                Object {
                  "bottom": -3,
                  "color": "var(--main-bg-color)",
                  "height": "20%",
                  "left": -100,
                  "position": "absolute",
                  "width": "calc(100vw + 101px)",
                  "zIndex": 1,
                }
              }
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polygon
                className="svg-header"
                points="0,100 100,0 100,100"
              />
            </svg>,
          ],
          "style": Object {
            "height": "40vh",
            "position": "relative",
            "width": "100vw",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "style": Object {
                "background": "url(./cake.jpg) no-repeat",
                "backgroundPosition": "center",
                "backgroundSize": "cover",
                "filter": "blur(1px)",
                "height": "inherit",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "article",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <polygon
                className="svg-header"
                points="0,100 100,0 100,100"
              />,
              "preserveAspectRatio": "none",
              "style": Object {
                "bottom": -3,
                "color": "var(--main-bg-color)",
                "height": "20%",
                "left": -100,
                "position": "absolute",
                "width": "calc(100vw + 101px)",
                "zIndex": 1,
              },
              "viewBox": "0 0 100 100",
              "xmlns": "http://www.w3.org/2000/svg",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "svg-header",
                "points": "0,100 100,0 100,100",
              },
              "ref": null,
              "rendered": null,
              "type": "polygon",
            },
            "type": "svg",
          },
        ],
        "type": "section",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <object
            data="./brand_logo.svg"
            style={
              Object {
                "height": "10em",
                "left": "2%",
                "position": "absolute",
                "top": "2%",
                "zIndex": 1,
              }
            }
            type="image/svg+xml"
          >
            <img
              alt="No SVG support"
              src="./cake.jpg"
            />
          </object>,
          <section
            style={
              Object {
                "height": "40vh",
                "position": "relative",
                "width": "100vw",
              }
            }
          >
            <article
              style={
                Object {
                  "background": "url(./cake.jpg) no-repeat",
                  "backgroundPosition": "center",
                  "backgroundSize": "cover",
                  "filter": "blur(1px)",
                  "height": "inherit",
                }
              }
            />
            <svg
              preserveAspectRatio="none"
              style={
                Object {
                  "bottom": -3,
                  "color": "var(--main-bg-color)",
                  "height": "20%",
                  "left": -100,
                  "position": "absolute",
                  "width": "calc(100vw + 101px)",
                  "zIndex": 1,
                }
              }
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polygon
                className="svg-header"
                points="0,100 100,0 100,100"
              />
            </svg>
          </section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="No SVG support"
              src="./cake.jpg"
            />,
            "data": "./brand_logo.svg",
            "style": Object {
              "height": "10em",
              "left": "2%",
              "position": "absolute",
              "top": "2%",
              "zIndex": 1,
            },
            "type": "image/svg+xml",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "No SVG support",
              "src": "./cake.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "object",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <article
                style={
                  Object {
                    "background": "url(./cake.jpg) no-repeat",
                    "backgroundPosition": "center",
                    "backgroundSize": "cover",
                    "filter": "blur(1px)",
                    "height": "inherit",
                  }
                }
              />,
              <svg
                preserveAspectRatio="none"
                style={
                  Object {
                    "bottom": -3,
                    "color": "var(--main-bg-color)",
                    "height": "20%",
                    "left": -100,
                    "position": "absolute",
                    "width": "calc(100vw + 101px)",
                    "zIndex": 1,
                  }
                }
                viewBox="0 0 100 100"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  className="svg-header"
                  points="0,100 100,0 100,100"
                />
              </svg>,
            ],
            "style": Object {
              "height": "40vh",
              "position": "relative",
              "width": "100vw",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "style": Object {
                  "background": "url(./cake.jpg) no-repeat",
                  "backgroundPosition": "center",
                  "backgroundSize": "cover",
                  "filter": "blur(1px)",
                  "height": "inherit",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "article",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <polygon
                  className="svg-header"
                  points="0,100 100,0 100,100"
                />,
                "preserveAspectRatio": "none",
                "style": Object {
                  "bottom": -3,
                  "color": "var(--main-bg-color)",
                  "height": "20%",
                  "left": -100,
                  "position": "absolute",
                  "width": "calc(100vw + 101px)",
                  "zIndex": 1,
                },
                "viewBox": "0 0 100 100",
                "xmlns": "http://www.w3.org/2000/svg",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "svg-header",
                  "points": "0,100 100,0 100,100",
                },
                "ref": null,
                "rendered": null,
                "type": "polygon",
              },
              "type": "svg",
            },
          ],
          "type": "section",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
